homeassistant/components/broadlink/switch.py

```diff
@@ -360,3 +364,48 @@ async def _async_send_packet(self, packet):
             _LOGGER.error("Failed to send packet: %s", err)
             return False
         return True
+
+
+class BroadlinkBG1Slot(BroadlinkSwitch):
+    """Representation of a Broadlink BG1 slot."""
+
+    def __init__(self, device, slot):
+        """Initialize the switch."""
+        super().__init__(device, 1, 0)
+        self._slot = slot
+        self._state = self._coordinator.data[f"pwr{slot}"]
+        self._device_class = DEVICE_CLASS_OUTLET
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the slot."""
+        return f"{self._device.unique_id}-s{self._slot}"
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return f"{self._device.name} S{self._slot}"
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of the switch."""
+        return False
+
+    @callback
+    def update_data(self):
+        """Update data."""
+        if self._coordinator.last_update_success:
+            self._state = self._coordinator.data[f"pwr{self._slot}"]
+        self.async_write_ha_state()
+
+    async def _async_send_packet(self, packet):
+        """Send a packet to the device."""
+        try:
+            set_state = partial(
```

Comment:
Please only wrap the line that can raise in the try... except block.

Reply:
```python3
        set_state = partial(self._device.api.set_state, **{f"pwr{self._slot}": packet})
        try:
            await self._device.async_request(set_state)
```

Reply:
Done. But out of curiosity @MartinHjelmare, why? As I understand it, the overhead of the `try` is negligible. Admittedly it's been 12+ years since I was doing active development, and that wasn't in python, but wrapping the entire block marks it as a related "transaction", and IMO improves readibility.

Reply:
We want the try scope to be minimal to make it clear what can raise. This improves readability. It's also safer to do so to avoid catching errors unintentionally.